python/ray/ml/tests/test_predictor.py

```diff
@@ -26,6 +35,23 @@ def test_serialization():
         ray.put(predictor)
 
 
+def test_batch_prediction():
+    batch_predictor = BatchPredictor.from_checkpoint(
+        Checkpoint.from_dict({"factor": 2.0}), DummyPredictor
+    )
+
+    test_dataset = ray.data.from_items([1.0, 2.0, 3.0, 4.0])
+    # Todo: Remove sorted after https://github.com/ray-project/ray/issues/23833 fixed
```

Comment:
There's an ordering bug tracked in https://github.com/ray-project/ray/issues/23833
Since this is a general error and not part of this PR, I'm relaxing the test constraints here.

cc @clarkzinzow can you remove this when https://github.com/ray-project/ray/issues/23833 is fixed?

Reply:
@krfricke fixing [PR](https://github.com/ray-project/ray/pull/23837) is out FYI!